packages/core/src/graph/create-nodes.ts

```diff
@@ -26,9 +26,27 @@ import { getWorkspaceScope } from '../generators/utils/get-scope';
 import type { PackageJson } from 'nx/src/utils/package-json';
 import { tryReadJsonFile } from '../generators/utils/try-read-json';
 
-export const projectFilePatterns = readConfig().inferProjects
-  ? ['*.csproj', '*.fsproj', '*.vbproj']
-  : [];
+let cachedProjectFilePatterns: string[] | null = null;
+
+export function getProjectFilePatterns(): string[] {
+  if (cachedProjectFilePatterns !== null) {
+    return cachedProjectFilePatterns;
+  }
+  try {
+    cachedProjectFilePatterns = readConfig().inferProjects
+      ? ['*.csproj', '*.fsproj', '*.vbproj']
+      : [];
+  } catch (error) {
+    console.warn('Failed to read nx-dotnet config:', error);
+    cachedProjectFilePatterns = [];
+  }
+  return cachedProjectFilePatterns;
+}
```

Comment:
We shouldn't cache this like this, if the daemon is enabled and users edit config this would require running nx reset to take effect, and readConfig is a small util function that shouldn't be causing issues.